test.js

```diff
@@ -586,3 +586,42 @@ if (Promise.prototype.finally) {
 		t.is(result.message, 'called');
 	});
 }
+
+test('cancel method kills the spawned process', t => {
+	const spawned = execa('node');
+	spawned.cancel();
+	t.true(spawned.killed);
+});
+
+test('result.canceled is false when spawned.cancel isn\'t called', async t => {
+	const spawned = execa('noop');
+	const result = await spawned;
+	t.false(result.canceled);
+});
+
+test('calling cancel method throws an error with message "Command was canceled"', async t => {
+	const spawned = execa('noop');
+	spawned.cancel();
+	const error = await t.throwsAsync(async () => {
+		await spawned;
+	});
+	t.regex(error.message, /Command\swas\scanceled/);
+});
+
+test('error.canceled is true when cancel method is used', async t => {
+	const spawned = execa('noop');
+	spawned.cancel();
+	const error = await t.throwsAsync(async () => {
+		await spawned;
+	});
+	t.true(error.canceled);
+});
+
+test('error.canceled is false when kill method is used', async t => {
```

Comment:
Except for comments above, tests look good now!

Just three extra tests that you might want to add: 
  - calling `spawned.cancel()` twice should have same behavior as calling it once.
  - calling `spawned.cancel()` on a child process that has already successfully exited
  - calling `spawned.cancel()` on a child process that has already been killed